packages/nimbus/src/components/money-input/money-input.tsx

```diff
@@ -351,8 +328,12 @@ export const MoneyInputComponent = (props: MoneyInputProps) => {
             placeholder={placeholder}
             autoFocus={autoFocus}
             size={size}
-            aria-label={ariaLabel}
-            aria-labelledby={ariaLabelledBy}
+            //base accessible name: "Amount"
+            aria-label={intl.formatMessage(messages.amountInputLabel)}
+            // accessible name when hasNoCurrencies=true: "<CURRENCY_CODE> Amount"
+            aria-labelledby={noCurrenciesLabelId}
+            // accessible name when high precision: "High Precision Amount"
+            aria-describedby={highPrecisionBadgeId}
```

Comment:
just...trust me on this...I spent a lot of time with accessible names and how different labelling strategies affect them - this is the way